pallets/spin-polkadot/src/lib.rs

```diff
@@ -1,289 +1,195 @@
-	pub struct FastchainInherentData {
-		pub validation_data: PersistedValidationData,
-		/// A storage proof of a predefined set of keys from the relay-chain.
-		///
-		/// Specifically this witness contains the data for:
-		///
-		/// - the current slot number at the given relay parent
-		/// - active host configuration as per the relay parent,
-		/// - the relay dispatch queue sizes
-		/// - the list of egress HRMP channels (in the list of recipients form)
-		/// - the metadata for the egress HRMP channels
-		pub relay_chain_state: sp_trie::StorageProof,
-		// /// Downward messages in the order they were sent.
-		// pub downward_messages: Vec<InboundDownwardMessage>,
-		// /// HRMP messages grouped by channels. The messages in the inner vec must be in order
-		// they /// were sent. In combination with the rule of no more than one message in a
-		// channel per block, /// this means `sent_at` is **strictly** greater than the previous
-		// one (if any). pub horizontal_messages: BTreeMap<ParaId, Vec<InboundHrmpMessage>>,
-	}
+	use super::*;
 
-	/// A GRANDPA justification for block finality, it includes a commit message
-	/// and an ancestry proof including all headers routing all precommit
-	/// target blocks to the commit target block. Due to the current voting
-	/// strategy the precommit targets should be the same as the commit target,
-	/// since honest voters don't vote past authority set change blocks.
-	///
-	/// See the [original reference](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2412-2/substrate/primitives/consensus/grandpa/src/lib.rs#L133)
-	#[derive(CloneNoBound, Encode, Decode, PartialEqNoBound, RuntimeDebugNoBound, TypeInfo)]
-	#[scale_info(skip_type_params(MaxVotesAncestries))]
-	pub struct GrandpaJustification<MaxVotesAncestries: Get<u32>, H: Header> {
-		pub round: u64,
-		// TODO: replace with bounded size data structure
-		pub commit: Commit<H>,
-		pub votes_ancestries: BoundedVec<H, MaxVotesAncestries>,
+	/// Stored configuration for the GRANDPA authority set that signs FastChain finality proofs.
+	#[derive(Clone, Encode, Decode, TypeInfo, PartialEq, Eq)]
+	pub struct AuthoritySetData {
+		pub set_id: SetId,
+		pub authorities: AuthorityList,
 	}
 
-	/// Alive message proof combining `FastchainInherentData` and
-	/// `GrandpaJustification`.
-	///
-	/// Should be sent from a fastchain node to the parachain SPIN pallet via an
-	/// extrinsic call.
-	#[derive(CloneNoBound, Encode, Decode, PartialEqNoBound, RuntimeDebugNoBound, TypeInfo)]
-	#[scale_info(skip_type_params(MaxVotesAncestries))]
-	pub struct AliveMessageProof<MaxVotesAncestries: Get<u32>, H: Header> {
-		pub fastchain_inherent_data: FastchainInherentData,
-		pub grandpa_justification: GrandpaJustification<MaxVotesAncestries, H>,
+	/// Metadata about the best FastChain block accepted on the parachain.
+	#[derive(Clone, Encode, Decode, TypeInfo, PartialEq, Eq, Default)]
+	pub struct FinalizedTarget<Hash> {
+		pub number: u32,
```

Comment:
Maybe something like this?

```suggestion
	pub struct FinalizedTarget<T: Config, Hash> {
		pub number: frame_system::pallet_prelude::BlockNumberFor<T>,
```